easybuild/easyblocks/e/elpa.py

```diff
@@ -176,6 +180,52 @@ def run_all_steps(self, *args, **kwargs):
 
         return super(EB_ELPA, self).run_all_steps(*args, **kwargs)
 
+    def configure_step(self):
+        """Configure step for ELPA"""
+
+        # Add nvidia GPU support if requested
+        cuda_root = get_software_root('CUDA')
+        self.log.info("Got cuda root: %s", cuda_root)
+        if cuda_root:
+            self.cfg.update('configopts', '--enable-nvidia-gpu')
+            self.cfg.update('configopts', '--with-cuda-path=%s' % cuda_root)
+            self.cfg.update('configopts', '--with-cuda-sdk-path=%s' % cuda_root)
+
+            cuda_cc = build_option('cuda_compute_capabilities') or self.cfg['cuda_compute_capabilities']
+            if not cuda_cc:
+                raise EasyBuildError('List of CUDA compute capabilities must be specified, either via '
+                                     'cuda_compute_capabilities easyconfig parameter or via '
+                                     '--cuda-compute-capabilities')
+
+            # ELPA's --with-NVIDIA-GPU-compute-capability only accepts a single architecture
+            if len(cuda_cc) == 1:
+                cuda_cc = cuda_cc[0]
+                cuda_cc_string = cuda_cc.replace('.', '')
+                self.cfg.update('configopts', '--with-NVIDIA-GPU-compute-capability=sm_%s' % cuda_cc_string)
+                self.log.info("Enabling nvidia GPU support for compute capability: %s", cuda_cc_string)
+                # There is a dedicated kernel for sm80, but only from version 2021.11.001 onwards
+                if float(cuda_cc) >= 8.0 and LooseVersion(self.version) >= LooseVersion('2021.11.001'):
+                    self.cfg.update('configopts', '--enable-nvidia-sm80-gpu')
+            else:
+                raise EasyBuildError('ELPA currently only supports specifying one architecture when '
+                                     'building. You specified cuda-compute-capabilities: %s', cuda_cc)
```

Comment:
reordering suggestion to reduce indentation a bit:
```suggestion
            if len(cuda_cc) > 1:
                raise EasyBuildError('ELPA currently only supports specifying one architecture when '
                                     'building. You specified cuda-compute-capabilities: %s', cuda_cc)
            cuda_cc = cuda_cc[0]
            cuda_cc_string = cuda_cc.replace('.', '')
            self.cfg.update('configopts', '--with-NVIDIA-GPU-compute-capability=sm_%s' % cuda_cc_string)
            self.log.info("Enabling nvidia GPU support for compute capability: %s", cuda_cc_string)
            # There is a dedicated kernel for sm80, but only from version 2021.11.001 onwards
            if float(cuda_cc) >= 8.0 and LooseVersion(self.version) >= LooseVersion('2021.11.001'):
                self.cfg.update('configopts', '--enable-nvidia-sm80-gpu')
```

Reply:
Lol, that's funny, I had that before, but decided I didn't like it because `float(cuda_cc)` really assumes `len(cuda_cc)==`1, and I was afraid that putting it apart further in the code would make it easier to make mistakes later on (i.e. someone changes the if-statement, and the assumption that `cuda_cc` is a single element is suddenly broken).

I'm ok in changing it again, but, just wondering what you think of my argument above. Am I overly careful here? Or is there something to be said for keeping the code indented, as it makes explicit that this only runs under that assumption of the `if`-statement?

Reply:
Ok, just checked with a colleague who I know is quite strict on style, and he prefers your approach: checking the 'type' early, throw an error if it's not ok, and afterwards just assume it's fine to keep indentation limited. I'll go ahead and adapt it :)